lib/logging/logging.go

```diff
@@ -135,13 +137,28 @@ func (rf *rollingFile) open() error {
 }
 
 func (rf *rollingFile) roll() error {
-	// FUTURE: Support more than one roll.
+	// Flush the current buffer
 	rf.bufWriter.Flush()
 	rf.file.Close()
-	archivedFile := rf.file.Name() + ".1"
-	// Remove old archive and copy over existing
-	os.Remove(archivedFile)
-	os.Rename(rf.file.Name(), archivedFile)
+
+	// Start from the last backup file and move everything back by 1 step
+	for i := rf.maxBackupFiles; i > 0; i-- {
+		var olderFile, newerFile string
+
+		if i == 1 {
+			olderFile = rf.file.Name() // Original file
+		} else {
+			olderFile = fmt.Sprintf("%s.%d", rf.file.Name(), i-1)
+		}
+		newerFile = fmt.Sprintf("%s.%d", rf.file.Name(), i)
+
+		// If the older file exists, rename it to the next higher numbered file
+		if _, err := os.Stat(olderFile); err == nil {
+			os.Rename(olderFile, newerFile)
```

Comment:
An error returned by `os.Rename` should be verified. Then, calling `os.Stat` would be unnecessary.
Probably,
```go
switch err := os.Rename(olderFile, newerFile); {
case err == nil, errors.Is(err, os.ErrNotExist):
    continue
case err != nil:
    // error handling
}
```